### ISSUE TITLE
Images on cloud version

### ISSUE THREAD

**Issue by dleute** [sulu-io/sulu-cloud#3]
I am exploring sulu on heroku and when I tried to upload images, they don't appear.

Like most cloud engines, the storage is ephemeral. So I'm guessing this has something to do with why they are not showing up. but even if it is an unrelated bug, it's still ephemeral.

Does sulu have a way to push images to an object store? This is how I handle that problem in my own symfony projects. On upload, they are manipulated if necessary and then transferred to S3. All the asset url's for all varieties of images are then pulled from an S3 url.

--Derrek

**Comment by wachterjohannes** [comment]
hmm @alexander-schranz already startet to implement the storage layer for media https://github.com/sulu-io/sulu/pull/1139 we should really try to finish that PR.

currently you could only use the config to use a dir which is a mounted reliable storage.

```
sulu_media:
    storage:
        local:
            path: '%kernel.root_dir%/../uploads/media'
```

maybe this will help you

**Comment by dleute** [comment]
I don't know if heroku has a mountable form of non-ephemeral storage. (this may also be a permissions issue. very little control there). I have yet to get a symfony project working on any cloud engine reliably due to these issues.

Changing the path likely wouldn't help (even if I was able to make uploads work). When the app shuts down, the media would be gone.

I like the storage engine solution!

**Comment by danrot** [comment]
AFAIC the ephemeral storage is the only one available in heroku, so we really need the storage engine solution. That would make Sulu really cloud ready :smiley:

**Comment by dleute** [comment]
Yep. I've done it on previous symfony apps. But it does represent all kinds of issues. I only ever did "just enough" to make the app work. Not really a generalized solution.

**Comment by dleute** [comment]
The biggest issue I've had is so many things that needed to use assets (like other bundles) don't have a way to hook in to a modular storage system. So even though I could push my own assets to an object store, other bundles couldn't. It was really frustrating. This is something that symfony should probably start absorbing at the core. That way people can just hook into an existing asset storage system everywhere.
